src/coreWebVitals.test.ts

```diff
@@ -0,0 +1,219 @@
+		const mockCallback = jest.fn();
+		initCoreWebVitals(
+			{ browserId: 'abc', pageViewId: '123' },
+			mockCallback,
+		);
+
+		setVisibilityState('hidden');
+		global.dispatchEvent(new Event('visibilitychange'));
+		global.dispatchEvent(new Event('pagehide'));
+
+		expect(mockCallback).toBeCalledTimes(2);
+		expect(mockCallback).toBeCalledWith(false);
+	});
+
+	it('only registers pagehide if document is visible', () => {
+		const mockCallback = jest.fn();
+		initCoreWebVitals(
+			{ browserId: 'abc', pageViewId: '123' },
+			mockCallback,
+		);
+
+		setVisibilityState('visible');
+		global.dispatchEvent(new Event('visibilitychange'));
+		global.dispatchEvent(new Event('pagehide'));
+
+		expect(mockCallback).toBeCalledTimes(1);
+		expect(mockCallback).toBeCalledWith(false);
+	});
+
+	it('triggers the callback with `true` if metrics will be sent', () => {
+		const mockCallback = jest.fn();
+		initCoreWebVitals(
+			{ browserId: 'abc', pageViewId: '123' },
+			mockCallback,
+		);
+
+		forceSendMetrics();
+		global.dispatchEvent(new Event('pagehide'));
+
+		expect(mockCallback).toBeCalledTimes(1);
+		expect(mockCallback).toHaveBeenLastCalledWith(true);
+	});
+
+	it('does not trigger a callback if none is passed', () => {
+		const mockCallback = jest.fn(); // won’t be used
+		const mockAddEventListener = jest.spyOn(global, 'addEventListener');
+		initCoreWebVitals({ browserId: 'abc', pageViewId: '123' });
+
+		setVisibilityState('hidden');
+		global.dispatchEvent(new Event('visibilitychange'));
+		global.dispatchEvent(new Event('pagehide'));
+
+		expect(mockCallback).not.toHaveBeenCalled();
+		expect(mockAddEventListener).toHaveBeenCalledTimes(2);
+	});
+});
+
+describe('roundWithDecimals', () => {
+	it.each([
+		[1, 3, 3],
+		[1, 10.0, 10],
+		[1, 10.3, 10.3],
+		[1, 2.5, 2.5],
+		[3, 0.001_234, 0.001],
+		[4, 0.001_234, 0.001_2],
+		[5, 100.102_030_405_060_708_090, 100.102_03],
+		[6, 12345.000_001_2, 12345.000_001],
+		[9, 199.001_002_003_456, 199.001_002_003],
+	])('With precision %s, %f becomes %f', (precision, before, after) => {
+		expect(roundWithDecimals(before, precision)).toBe(after);
+	});
+
+	it('should handle default precision = 6', () => {
+		const [before, after] = [12345.000_001_2, 12345.000_001];
+		expect(roundWithDecimals(before)).toBe(after);
+		expect(roundWithDecimals(before, 6)).toBe(after);
+	});
+});
+
+describe('sendData', () => {
+	beforeEach(() => {
+		jest.resetModules();
+		window.location.hash = '';
+		resetShouldForceMetrics();
+	});
+
+	const browserId = String(defaultCoreWebVitalsPayload.browser_id);
+	const pageViewId = String(defaultCoreWebVitalsPayload.page_view_id);
+
+	it('should send data if in sample', () => {
+		mockMath.mockReturnValueOnce(0.1 / 100);
+		initCoreWebVitals({ browserId, pageViewId });
+
+		expect(sendData()).toBe(true);
+	});
+
+	it('should not send data if not in sample', () => {
+		initCoreWebVitals({ browserId, pageViewId });
+
+		expect(sendData()).toBe(false);
+	});
```

Comment:
What's the difference between these two tests? I can see one has a mock for a call which means the metrics are not sent, but other one expects the results to never be sent but for no obvious reason

Is it possible to make it more explicit why one test expects one things and the other as part of the test code itself?